components/raftstore/src/store/util.rs

```diff
@@ -349,9 +352,16 @@ pub fn check_flashback_state(
     }
     let is_flashback_request = WriteBatchFlags::from_bits_truncate(req.get_header().get_flags())
         .contains(WriteBatchFlags::FLASHBACK);
-    // If the region is in the flashback state, the only allowed request is the
-    // flashback request itself.
+    // If the region is in the flashback state:
+    //   - A request with flashback flag will be allowed.
+    //   - A read request whose `read_ts` is smaller than `flashback_start_ts` will
+    //     be allowed.
     if is_in_flashback && !is_flashback_request {
+        if let Ok(read_ts) = decode_u64(&mut req.get_header().get_flag_data()) {
+            if read_ts != 0 && read_ts < flashback_start_ts {
+                return Ok(());
+            }
+        }
         return Err(Error::FlashbackInProgress(region_id));
     }
     // If the region is not in the flashback state, the flashback request itself
```

Comment:
I have this idea, but I don't know if it will be more complicated :)
```
match (is_in_flashback, is_flashback_request) {
    (true, false) => match decode_u64(&mut req.get_header().get_flag_data()) {
        Ok(read_ts) if read_ts != 0 && read_ts < flashback_start_ts => return Ok(()),
        _ => return Err(Error::FlashbackInProgress(region_id)),
    },
    (false, true) if !skip_not_prepared => return Err(Error::FlashbackNotPrepared(region_id)),
    _ => (),
}
Ok(())
```

Reply:
I prefer just using the simple `if` statements here. It's clear enough 🤔.